### PR TITLE
Add Pillar SQLCipher module

### LINKED ISSUE #29781: Add Pillar SQLCipher module
SQLCipher is an open source extension to SQLite that provides transparent 256-bit AES encryption of database files.

* https://www.zetetic.net/sqlcipher/
* https://pypi.python.org/pypi/pysqlcipher

It is good to have support for retrieving Pillar data via queries to SQLCipher databases.

```
pillar:
  sqlcipher:
    database: '/var/lib/salt/pillar-sqlcipher.db'
    pass: strong_pass_phrase
    timeout: 5.0

ext_pillar:
  - sqlcipher:
      fromdb:
        query: 'SELECT col1,col2,col3,col4,col5,col6,col7
                  FROM some_random_table
                 WHERE minion_pattern LIKE ?'
        depth: 5
        as_list: True
        with_lists: [1,3]
```

### PR DESCRIPTION
**Pull request by akissa**
Add support for retrieving Pillar data via a queries to SQLCipher databases.

Fixes #29781

### CODE REVIEW

**Comment by cachedout** [comment]
Wow! This is really clean code. Very impressive. This will make an outstanding addition to Salt! Thanks very much for this contribution. I'm really excited to use it.

File: salt/pillar/sqlcipher.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+'''
+Retrieve Pillar data by running a SQLCipher query
+
```
Comment (jfindlay) [inline salt/pillar/sqlcipher.py]: This needs a `.. versionadded:: Boron` here.

**Comment by akissa** [comment]
@jfindlay Added in PR https://github.com/saltstack/salt/pull/29805

**Comment by jfindlay** [comment]
Excellent, thanks @akissa.

**Pull request merged**